mistralrs-core/src/pipeline/mod.rs

```diff
@@ -244,23 +244,155 @@ pub enum ModelKind {
     },
 }
 
-impl Display for ModelKind {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+// TODO: Future replacement for `ModelKind` above:
+#[derive(Default, derive_more::From, strum::Display)]
+pub enum ModelKindB {
```

Comment:
**Original comment removed from commit:**
- No need to `impl Display`, strum can handle derive `Display` with customized presentation.
- DRY via compound variants (_minimize growing permutations of adapters + formats_)
- Can derive `strum::EnumIs` for parent type to impl `is_quantized()` / `is_adapter()` methods that call
   - `kind.is_quantized() || kind.is_adapter_quantized()`
   - `kind.is_adapter() || kind.is_adapter_quantized()`
- Alternatively derive `strum::EnumProperty` to assign similar information to variants and query via `get_str(prop)`

---

Due to the `Speculative` variant and advice to return a Vec of options instead, the `strum::EnumIs` was only used on the sub variants (`QuantizationKind` / `AdapterKind`). As the logic for `is_quantized()` need to be a bit more complicated.